extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/client/IndexerWorkerManagerClient.java

```diff
@@ -65,10 +65,13 @@ public Map<String, TaskStatus> statuses(Set<String> taskIds)
   @Override
   public TaskLocation location(String workerId)
   {
-    final TaskStatusResponse response = FutureUtils.getUnchecked(overlordClient.taskStatus(workerId), true);
+    final TaskStatus response = FutureUtils.getUnchecked(
+        overlordClient.taskStatuses(ImmutableSet.of(workerId)),
+        true
+    ).get(workerId);
 
-    if (response.getStatus() != null) {
-      return response.getStatus().getLocation();
```

Comment:
We need the same change in `SpecificTaskServiceLocator`, to use `taskStatuses` instead of `taskStatus`.

Reply:
Done